### PR TITLE
dev-python/pyzbar: Introducing package

### PR DESCRIPTION
**Pull request by bucefal91**
Introducing the new package into Gentoo.

Closes: https://bugs.gentoo.org/655078
Package-Manager: Portage-2.3.33, Repoman-2.3.9

### CODE REVIEW

**Comment by a17r** [comment]
Since this depends on `zbar`, would you also be interested in picking up https://bugs.gentoo.org/641920?

**Comment by bucefal91** [comment]
Alright. I am a beginner with ebuilds but yep, I'll have a look into that bug. I will file one more dev-python/* pull request and will switch to `zbar` :)

**Comment by a17r** [comment]
Package is installing `tests` files and missing at least the following dependencies per requirements.txt:

- `dev-python/pillow`
- `dev-python/enum34`
- `dev-python/pathlib`

@gentoo/python maybe you can take a look at it

**Comment by bucefal91** [comment]
Yes, I documented the original dependency just from README.txt and didn't look into requirements.pip. So I processed those too (I believe). While doing those dependencies a few points arose in my head:
* instead of "dev-python/pathlib" I found "virtual/python-pathlib". I assume it's better to depend on the virtual/* , isn't it?
* To be honest so far I couldn't figure anything better than rm -rf tests in src_install (was suggested by @a17r )
* Is it ok to do conditionals on python version this way:
```
python_targets_python2_7? (
  =dev-python/enum34-1.1.6
  virtual/python-pathlib
)
```
or I was supposed to use some of the fancy functions from distutils-r1?

**Comment by floppym** [comment]
> Package is installing tests files

The tests are installed in a package sub-directory, so there is no problem here.

File: dev-python/pyzbar/pyzbar-0.1.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="
+	media-gfx/zbar
+	dev-python/pillow
+	dev-python/numpy
+	python_targets_python2_7? (
```
Comment (floppym) [inline dev-python/pyzbar/pyzbar-0.1.5.ebuild]: Get rid of the `python_targets_python2_7` conditional, and depend on `virtual/python-enum34` and `virtual/python-pathlib`.

File: dev-python/pyzbar/pyzbar-0.1.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+SRC_URI="https://github.com/NaturalHistoryMuseum/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="
+	media-gfx/zbar
+	dev-python/pillow
+	dev-python/numpy
```
Comment (floppym) [inline dev-python/pyzbar/pyzbar-0.1.5.ebuild]: numpy is only required for testing.

**Comment by floppym** [comment]
Also, your python dependencies are missing `[${PYTHON_USEDEP}]`.

**Comment by bucefal91** [comment]
@floppym :) I've gone through your feedback. In particular about the tests - I was able to execute them and all the tests pass successfully.

File: dev-python/pyzbar/pyzbar-0.1.5.ebuild [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
+inherit distutils-r1
+
+DESCRIPTION="Read one-dimensional barcodes and QR codes from Python."
```
Comment (mgorny) [inline dev-python/pyzbar/pyzbar-0.1.5.ebuild]: This is not a proper sentence, so no dot at the end.

File: dev-python/pyzbar/pyzbar-0.1.5.ebuild [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		dev-python/numpy[${PYTHON_USEDEP}]
+		dev-python/mock[${PYTHON_USEDEP}]
+	)
+"
+
+python_test() {
+	nosetests -v || die
```
Comment (mgorny) [inline dev-python/pyzbar/pyzbar-0.1.5.ebuild]: You don't depend on nose. Also, it doesn't seem that the tests actually require nose, so please run them via built-in unittest if possible.

https://wiki.gentoo.org/wiki/Project:Python/Tests

File: dev-python/pyzbar/pyzbar-0.1.5.ebuild [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+SRC_URI="https://github.com/NaturalHistoryMuseum/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+RDEPEND="
+	media-gfx/zbar
+	dev-python/pillow[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/pyzbar/pyzbar-0.1.5.ebuild]: Please sort deps lexically.

File: dev-python/pyzbar/pyzbar-0.1.5.ebuild [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	virtual/python-enum34[${PYTHON_USEDEP}]
+	virtual/python-pathlib[${PYTHON_USEDEP}]
+"
+
+DEPEND="
+	${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		dev-python/numpy[${PYTHON_USEDEP}]
+		dev-python/mock[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/pyzbar/pyzbar-0.1.5.ebuild]: It uses `unittest.mock` when available, so this is only required for old Pythons. I've just pushed `virtual/python-unittest-mock` for you.

https://wiki.gentoo.org/wiki/Project:Python/Virtuals#Virtual_packages_for_backports

**Comment by bucefal91** [comment]
@mgorny Hello! :) I've processed your feedback. And you were right, the tests ran successfully with unittest. I picked nose because I found the upstream was mentioning it in their documentation.

**Comment by mgorny** [comment]
Thanks. I've squashed the commits, updated the commit message and I'll merge now.

As for tests, don't worry. It's common for upstreams to use py.test or nosetests to run tests because they're featureful and colorful. However, in our case we really want to go for least dependencies.

**Pull request closed without merging**